app/controllers/subscriptions_controller.rb

```diff
@@ -15,6 +15,7 @@ def new
 
     if params[:email]
       @email = params[:email]
+      @subscription = Subscription.find_by(email: @email)
```

Comment:
Ah I didn't think of that :+1:

People will most likely be coming from an alert email if they see that, would it be ok to use the alert.confirm_id as a way to hide this, as we do for the unsubscribe and area actions on alerts? Then you need the person's email and an email that's been sent to them.